drv/sidecar-seq-api/src/lib.rs

```diff
@@ -44,4 +50,58 @@ pub enum TofinoSequencerPolicy {
     RestartOnFault = 2,
 }
 
+#[derive(
+    Copy, Clone, Debug, PartialEq, Eq, Deserialize, Serialize, SerializedSize,
+)]
+#[repr(C)]
+pub struct FanModulePresence(pub [bool; NUM_FAN_MODULES]);
+
+/// Four fan modules exist on sidecar, each with two fans.
+///
+/// The SP applies control at the individual fan level. Power control and
+/// status, module presence, and module LED control exist at the module level.
+#[derive(Copy, Clone, Debug, PartialEq, Eq, FromPrimitive, AsBytes)]
+#[repr(u8)]
+pub enum FanModuleIndex {
```

Comment:
After a bit of thought, I have two suggestions:

- Move `FanModuleIndex` into `drv/sidecar-mainboard-controller/src/fan_modules.rs`.  This would also let us change the function signatures of methods in `FanModules` to take `idx: FanModuleIndex` instead of `idx: u8`.
- Lose all of the `impl From` below.  We've already got `from_usize` because we derived `FromPrimitive`, and since this is a simple `enum`, we can go to numerical type with just `idx as usize`.

I made these changes locally to test them out; you're welcome to borrow [the diff](https://gist.github.com/mkeeter/c43be110fa437f7ebb67752334963968).

Reply:
Another good suggestion, and thanks for the diff!